modules/reindex/src/test/java/org/elasticsearch/index/reindex/ReindexFromRemoteWithAuthTests.java

```diff
@@ -129,9 +138,21 @@ public void testReindexWithBadAuthentication() throws Exception {
      * Plugin that demands authentication.
      */
     public static class TestPlugin extends Plugin implements ActionPlugin {
+
+        private final SetOnce<ReindexFromRemoteWithAuthTests.TestFilter> testFilter = new SetOnce<>();
+
+        @Override
+        public Collection<Object> createComponents(Client client, ClusterService clusterService, ThreadPool threadPool,
+                                                   ResourceWatcherService resourceWatcherService, ScriptService scriptService,
+                                                   NamedXContentRegistry xContentRegistry, Environment environment,
+                                                   NodeEnvironment nodeEnvironment, NamedWriteableRegistry namedWriteableRegistry) {
+            testFilter.set(new ReindexFromRemoteWithAuthTests.TestFilter(threadPool));
```

Comment:
The purpose of createComponents is to create services. This creates an unenforcible ordering dependency between creating these services and calling `getActionFilters()`.  `getActionFilters` should take whatever is necessary to creation action filters.

Reply:
`getActionFilters` is meant to be used by plugins, for which we don't know about the parameters that they require. We have the same situation for a lot of other methods (e.g. getRestHandlers). In case of incorrect ordering, our tests will blow up.